compressai/latent_codecs/hyper.py

```diff
@@ -68,13 +68,17 @@ class HyperLatentCodec(LatentCodec):
     h_a: nn.Module
     h_s: nn.Module
 
-    def __init__(self, N: int, **kwargs):
+    def __init__(
+        self,
+        entropy_bottleneck: Optional[EntropyBottleneck] = None,
+        h_a: Optional[nn.Module] = None,
+        h_s: Optional[nn.Module] = None,
+        **kwargs,
+    ):
         super().__init__()
-        self._kwargs = kwargs
-        self.N = N
-        self._setdefault("entropy_bottleneck", lambda: EntropyBottleneck(N))
-        self._setdefault("h_a", nn.Identity)
-        self._setdefault("h_s", nn.Identity)
+        self.entropy_bottleneck = entropy_bottleneck or EntropyBottleneck(0)
```

Comment:
What is the rationale to init an entropybottleneck with 0 channels, can you extend it later?

In gainhyperlatentcodec, you init with 
self.entropy_bottleneck = entropy_bottleneck or EntropyBottleneck()

Reply:
You're right; this was a bit silly. I wanted it to type-check. Possible alternatives:

1. Force callers to provide `entropy_bottleneck`:

```python
assert entropy_bottleneck is not None
self.entropy_bottleneck = entropy_bottleneck
```

2. Restore the `z_channels` argument:

```python
def __init__(
    self,
    z_channels: Optional[int] = None,
    entropy_bottleneck: Optional[EntropyBottleneck] = None,
):
    self.entropy_bottleneck = entropy_bottleneck or EntropyBottleneck(z_channels)
```

Whichever seems least confusing / easy to use.